stylus-sdk/src/abi/mod.rs

```diff
@@ -76,6 +76,15 @@ where
     /// A fallback function can be declared as payable. If not payable, then any transactions
     /// that trigger a fallback with value attached will revert.
     fn fallback(storage: &mut S, calldata: &[u8]) -> Option<ArbResult>;
+
+    /// The router_entrypoint calls the constructor  when the selector is CONSTRUCTOR_SELECTOR.
+    /// The implementation should: decode the calldata and pass the parameters to the user-defined
+    /// constructor; and call internal::constructor_guard to ensure it is only executed once.
+    /// The constructor is assumed to be payable and doesn't need a #[paylable] annotation.
```

Comment:
```suggestion
    /// The constructor is assumed to be payable and doesn't need a #[payable] annotation.
```

Reply:
Is the same true for Solidity? I thought that one needed to declare constructors as payable explicitly. What if users don't want a default payable behavior? Was unable to find many sources on the matter, though

Reply:
I just tested solidity here, and the constructor requires the payable annotation to receive funds. I will fix it on our side to mimic solidity.